### PR TITLE
feat(perfs): Early return on expensive calls

### PR DESCRIPTION
**Pull request by Rushaway**

### CODE REVIEW

File: src/addons/sourcemod/scripting/zr/event.inc [hunk 1]
```diff
@@ -179,6 +179,11 @@ public Action EventPlayerSpawn(Handle event, const char[] name, bool dontBroadca
     // Get all required event info.
     int index = GetClientOfUserId(GetEventInt(event, "userid"));
 
+    // CS:S has a bug: It never verify if client is in a valid team before fire player_spawn event.
+    int iTeam = GetClientTeam(index);
+    if (iTeam == CS_TEAM_NONE || iTeam == CS_TEAM_SPECTATOR)
```
Comment (maxime1907) [inline src/addons/sourcemod/scripting/zr/event.inc:184]: not sure that its a good idea since some functions do not require the player to be in a team?

Comment (Rushaway) [reply src/addons/sourcemod/scripting/zr/event.inc:184]: They will be fired anyway when player will join a team and get spawned.

Comment (maxime1907) [reply src/addons/sourcemod/scripting/zr/event.inc:184]: so whats the purpose of this function if it gets fired after it anyway?

Comment (Rushaway) [reply src/addons/sourcemod/scripting/zr/event.inc:184]: I performed tests and tests again. This PR introduce a good improvement.
It prevent useless code to be performed. 
Plus it prevent the duplication of some messages based on EventPlayerSpawn

Comment (maxime1907) [reply src/addons/sourcemod/scripting/zr/event.inc:184]: ok we can give it a try then but if it introduces some weird behaviors we can just revert

**Pull request merged**